flow/internal/workflow.go

```diff
@@ -33,6 +34,11 @@ func GetWorkflowStatus(ctx context.Context, pool shared.CatalogPool,
 	err := pool.QueryRow(ctx, "SELECT status FROM flows WHERE workflow_id = $1", workflowID).Scan(&flowStatus)
 	if err != nil || flowStatus == protos.FlowStatus_STATUS_UNKNOWN {
 		if err != nil {
+			if errors.Is(err, pgx.ErrNoRows) {
+				slog.Warn("workflowId not found in catalog, will raise an error",
+					slog.String("workflowId", workflowID))
+				return protos.FlowStatus_STATUS_UNKNOWN, fmt.Errorf("workflowId not found in catalog: %w", err)
```

Comment:
If only maintenance can hit this I think it's fine to return `STATUS_RUNNING` and log a warning; this can lead to multiple alerts at once and I think we can investigate it without blocking maintenance